tests/test_expressions.py

```diff
@@ -451,6 +451,12 @@ def test_cast(self):
         self.assertEqual(len(list(recast.find_all(exp.Cast))), 2)
         self.assertEqual(recast.sql(), "CAST(CAST(x AS DATE) AS VARCHAR)")
 
+        # check that dialect is used when casting strings
+        self.assertEqual(
+            exp.cast("x", to="regtype", dialect="postgres").sql(), "CAST(x AS REGTYPE)"
+        )
+        self.assertEqual(exp.cast("`x`", to="date", dialect="hive").sql(), 'CAST("x" AS DATE)')
```

Comment:
these will fail without your fix?

Reply:
Yes top is for the parsing of data type and bottom is for parsing of column (due to weird quotes in hive). Both will fail without fix.